### PR TITLE
[Windows] Move to Visual Studio 2019

### PR DESCRIPTION
**Pull request by loic-sharma**
*This reapplies https://github.com/flutter/engine/pull/36538 which was reverted by https://github.com/flutter/engine/pull/36606 as it uncovered https://github.com/flutter/flutter/issues/113728.*

Use Visual Studio 2019 to build the engine.

This is a requirement to roll ANGLE as it is affected by an STL bug if built using Visual Studio 2017: [angleproject#7693](https://bugs.chromium.org/p/angleproject/issues/detail?id=7693)

Depends on: https://github.com/flutter/buildroot/pull/643

Part of: https://github.com/flutter/flutter/issues/110948

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: DEPS [hunk 1]
```diff
@@ -113,7 +113,7 @@ allowed_hosts = [
 ]
 
 deps = {
-  'src': 'https://github.com/flutter/buildroot.git' + '@' + 'a680bb1c63e6ae1831cb672db029e9842ec4b905',
+  'src': 'https://github.com/loic-sharma/flutter-buildroot.git' + '@' + 'ab9b11e999e48ba0b6769f42c14aafa437e1c417',
```
Comment (loic-sharma) [inline DEPS:116]: I'll update this after https://github.com/flutter/buildroot/pull/643 is merged

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Review by yaakovschectman — APPROVED** [review]
LGTM once you update DEPS

**Review by cbracken — APPROVED** [review]
lgtm

**Pull request merged**